### PR TITLE
Add Match 

### PR DESCRIPTION
**Pull request by dutor**
Depends on vesoft-inc/nebula-common#267

### CODE REVIEW

File: src/validator/MatchValidator.cpp [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+    scanInfo_.filter = indexFilter;
+    scanInfo_.schemaId = head.tid;
+
+    return Status::OK();
+}
+
+
+Expression* MatchValidator::createSubFilter(const std::string &alias,
+        const MapExpression *map) const {
+    DCHECK_NOTNULL(map);
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:253]: should not use `DCHECK_NOTNULL` to check variable nullable in one line. Replace it with `DCHECK(map != nullptr);` or `auto items = DCHECK_NOTNULL(map)->items();`

File: src/validator/MatchValidator.cpp [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    auto *clause = sentence->where();
+    if (clause == nullptr) {
+        return Status::OK();
+    }
+    auto *filter = clause->filter();
+    auto kind = filter->kind();
+    // TODO(dutor) Find a better way to identify where an expr is a boolean one
+    if (kind == Expression::Kind::kLabel ||
+            kind == Expression::Kind::kLabelAttribute) {
+        return Status::SemanticError("Filter should be a bllean expression");
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:503]: typo: boolean

File: src/validator/MatchValidator.cpp [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+namespace nebula {
+namespace graph {
+
+Status MatchValidator::toPlan() {
+    auto status = Status::OK();
+    do {
+        status = buildScanNode();
+        if (!status.ok()) {
+            break;
+        }
```
[the remark this answers is not in the document]
Comment (dutor) [reply src/validator/MatchValidator.cpp:20]: roger that.

File: src/validator/MatchValidator.cpp (hunk 1 above)
Comment (dutor) [reply src/validator/MatchValidator.cpp:253]: thanks

File: src/validator/MatchValidator.cpp (hunk 2 above)
Comment (dutor) [reply src/validator/MatchValidator.cpp:503]: thanks

**Review by yixinglu — DISMISSED** [review]
LGTM

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Review by yixinglu — APPROVED** [review]

**Pull request merged**